File: backend/models/Store.go
package models

import (
	"gorm.io/gorm"
)

type Store struct {
	gorm.Model
	MerchantID  uint    `json:"merchant_id" gorm:"not null;index"`            // ID du commerçant (clé étrangère)
	Name        string  `json:"name" gorm:"type:varchar(255);not null"`       // Nom du restaurant (obligatoire)
	Latitude    float64 `json:"latitude" gorm:"type:decimal(10,8);not null"`  // Latitude (format décimal)
	Longitude   float64 `json:"longitude" gorm:"type:decimal(11,8);not null"` // Longitude (format décimal)
	Address     string  `json:"address" gorm:"type:text;not null"`            // Adresse complète
	City        string  `json:"city" gorm:"type:varchar(100);not null"`       // Ville
	PostalCode  string  `json:"postal_code" gorm:"type:varchar(10);not null"` // Code postal (limité à 10 caractères pour compatibilité internationale)
	PhoneNumber string  `json:"phone_number" gorm:"type:varchar(15)"`         // Numéro de téléphone (optionnel, max 15 caractères)
	Rating      float64 `json:"rating" gorm:"default:0.00"`                   // Note moyenne (sur 5)
	CategoryID  uint    `json:"category_id" gorm:"not null;index"`            // ID de la catégorie (clé étrangère)

	Merchant Merchant `json:"merchant" gorm:"foreignKey:MerchantID;constraint:OnDelete:CASCADE"` // Relation avec Merchant (clé étrangère)
	Category Category `json:"category" gorm:"foreignKey:CategoryID"`                             // Relation avec Category (clé étrangère)
}

File: frontend/lib/models/store.dart
class Store {
  final int id;
  final String name;
  final String address;
  final String city;
  final String postalCode;
  final String phoneNumber;
  final int categoryId;
  final double? latitude;
  final double? longitude;
  final double? rating;

  Store({
    required this.id,
    required this.name,
    required this.address,
    required this.city,
    required this.postalCode,
    required this.phoneNumber,
    required this.categoryId,
    this.latitude,
    this.longitude,
    this.rating,
  });

  factory Store.fromJson(Map<String, dynamic> json) {
    return Store(
      id: json['ID'] ?? 0,
      name: json['name'] ?? '',
      address: json['address'] ?? '',
      city: json['city'] ?? '',
      postalCode: json['postal_code'] ?? '',
      phoneNumber: json['phone_number'] ?? '',
      categoryId: json['category_id'] ?? 0,
      latitude: json['latitude']?.toDouble(),
      longitude: json['longitude']?.toDouble(),
      rating: json['rating']?.toDouble(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'address': address,
      'city': city,
      'postal_code': postalCode,
      'phone_number': phoneNumber,
      'category_id': categoryId,
      'latitude': latitude,
      'longitude': longitude,
      'rating': rating,
    };
  }
}